### PR TITLE
Fixes #20265 - Add UEFI Grub(2) templates for SUSE

### PR DESCRIPTION
**Pull request by tux93**

### CODE REVIEW

**Comment by tbrisker** [comment]
ok to test

**Comment by ezr-ondrej** [comment]
@sbernhard and this one needs testing as well, would be nice to test it together with #8638.

**Comment by ares** [comment]
It would be great to also add a description in metadata section, please see https://github.com/theforeman/foreman/pull/8669 for examples.

File: app/views/unattended/provisioning_templates/PXEGrub2/autoyast_default_pxegrub.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+default=0
+timeout=<%= host_param('loader_timeout') || 10 %>
+<% http_proxy_string = host_param('http-proxy') ? "proxy=http://" + host_param('http-proxy') + ":" + host_param('http-proxy-port') : '' -%>
+
+title <%= template_name %>
+  root (nd)
+  kernel (nd)/../<%= @kernel %> ramdisk_size=65536 install=<%= @mediapath %> autoyast=<%= foreman_url('provision') %> textmode=1 <%= http_proxy_string %>
+  initrd (nd)/../<%= @initrd %>
+
+<%= snippet_if_exists(template_name + " custom menu") %>
```
Comment (lzap) [inline app/views/unattended/provisioning_templates/PXEGrub2/autoyast_default_pxegrub.erb]: Wrong directory, have you tested this? Grub1 is rarely used, we have it for RHEL6 provisioning but this is old stuff.

File: app/views/unattended/provisioning_templates/PXEGrub2/autoyast_default_pxegrub2.erb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  kernel_options = "ramdisk_size=65536 install=#{@mediapath} autoyast=#{foreman_url('provision')} textmode=1 #{http_proxy_string}"
+-%>
+
+set default=<%= host_param('default_grub_install_entry') || 0 %>
+set timeout=<%= host_param('loader_timeout') || 10 %>
+
+menuentry '<%= template_name %>' {
+  <%= linuxcmd %> <%= @kernel %> <%= kernel_options %>
+  <%= initrdcmd %> <%= @initrd %>
+}
```
Comment (lzap) [inline app/views/unattended/provisioning_templates/PXEGrub2/autoyast_default_pxegrub2.erb:41]: This is good, is this tested?

The rest is a bit crazy stuff for UEFI HTTP Boot, have you tried this? If you just copy pasted it here, I would rather not add it for now, until someone needs it.

File: app/views/unattended/provisioning_templates/PXEGrub2/autoyast_default_pxegrub.erb (hunk 1 above)
Comment (tux93) [reply app/views/unattended/provisioning_templates/PXEGrub2/autoyast_default_pxegrub.erb]: Moved to the correct directory

As for testing: my focus was on the Grub2 template but I can get caught up with a test of this one as well

File: app/views/unattended/provisioning_templates/PXEGrub2/autoyast_default_pxegrub2.erb (hunk 2 above)
Comment (tux93) [reply app/views/unattended/provisioning_templates/PXEGrub2/autoyast_default_pxegrub2.erb:41]: We did test this before opening the MR, it is what we currently use for UEFI environments

**Review by sbernhard — APPROVED** [review]

**Review by lzap — APPROVED** [review]
Thanks!

**Pull request merged**